newmiddleware/index.js

```diff
@@ -12,6 +16,25 @@ const getAuthMiddleware = ({ identityAuthService }) => {
   return authMiddleware
 }
 
+const formSGService = new FormSGService()
+const formSGMiddleware = new FormSGMiddleware({ formSGService })
+
+/**
+ * Handles FormSG response authentication and decryption. 
+ * 
+ * To be inserted before routes requiring FormSG response data - e.g. 
+ * router.post(
+      "/",
+      attachFormSGHandler(<formKey>),
+      actualRoute
+    )
+ *
+ * Retrieve form data from res.locals.submission.
+ */
+const attachFormSGHandler = (formKey) =>
+  formSGMiddleware.authenticateAndDecrypt({ formKey })
+
```

Comment:
double checking here - why not just inline the `authenticateAndDecrypt` method here? it seems to be just a wrapper over the underlying two methods + calling `express.json`. i don't think i'm understanding the benefit of the middleware

Reply:
As discussed offline, the primary reason for the middleware/service structure is consistency with the remainder of our codebase, as well as dependency injection for easier testing. As a compromise, since the formsgMiddleware is currently only serving as a thin wrapper, we are opting to move this into newmiddleware/index instead, as it is not a particularly complicated method.